src/client/testing/page-objects/view-availability-appointment-pages/week-view-availability-page.ts

```diff
@@ -0,0 +1,250 @@
+import { type Locator, type Page, expect } from '@playwright/test';
+import RootPage from '../root';
+
+export type dayOverview = {
+  header: string;
+  services: serviceOverview[];
+  totalAppointments: number;
+  booked: number;
+  unbooked: number;
+};
+
+export type serviceOverview = {
+  sessionTimeInterval: string;
+  serviceName: string;
+  booked: number;
+  unbooked: number;
+};
+
+export default class WeekViewAvailabilityPage extends RootPage {
+  readonly nextButton: Locator;
+  readonly previousButton: Locator;
+  readonly backToMonthButton: Locator;
+  readonly sessionSuccessMsg: Locator;
+  readonly viewDailyAppointmentButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+
+    this.nextButton = page.getByRole('link', {
+      name: 'Next',
+    });
+    this.previousButton = page.getByRole('link', {
+      name: 'Previous',
+    });
+    this.backToMonthButton = page.getByRole('link', {
+      name: 'Back to month view',
+    });
+    this.sessionSuccessMsg = page.getByText(
+      'You have successfully created availability for the current site.',
+    );
+    this.viewDailyAppointmentButton = page.getByRole('link', {
+      name: 'View daily appointments',
+    });
+  }
+
+  async verifyViewNextWeekButtonDisplayed() {
+    await expect(this.nextButton).toBeEnabled();
+  }
+
+  async verifyAllDayCardInformationDisplayedCorrectly(
+    expectedDayOverviews: dayOverview[],
+  ) {
+    for (let i = 0; i < expectedDayOverviews.length; i++) {
+      const dayOverview = expectedDayOverviews[i];
+
+      const cardDiv = this.page
+        .getByRole('heading', {
+          name: dayOverview.header,
+        })
+        .locator('..');
```

Comment:
Likewise think we could use filter/has here to traverse the tree in just one direction rather than bouncing up